packages/database/lib/migrations/20240708161616_index_sync_configs.cjs

```diff
@@ -0,0 +1,11 @@
+exports.config = { transaction: false };
+
+exports.up = async function (knex, _) {
+    await knex.schema.raw(
+        `CREATE INDEX CONCURRENTLY idx_nango_sync_configs_sync_name_active ON _nango_sync_configs USING BTREE (sync_name) WHERE active = true`
```

Comment:
I think we can improve even further with
`(type, sync_name) WHERE active = true and delete = false` and even further with `(type, sync_name, nango_config_id) WHERE active = true and delete = false`

Reply:
I tried your first option and it makes very little difference so I went for the simplest index. Here is the query plan with your second option: 
```
Nested Loop  (cost=0.84..10.41 rows=1 width=235)
  ->  Nested Loop Left Join  (cost=0.56..5.01 rows=1 width=107)
        ->  Index Scan using idx_connectionid_name_where_deleted on _nango_syncs  (cost=0.28..2.50 rows=1 width=87)
              Index Cond: (nango_connection_id = 20202)
        ->  Index Scan using idx_sync_id_active_true on _nango_active_logs  (cost=0.28..2.50 rows=1 width=36)
              Index Cond: (sync_id = _nango_syncs.id)
              Filter: ((type)::text = 'sync'::text)
  ->  Index Scan using idx_nango_sync_configs_composite_2 on _nango_sync_configs  (cost=0.28..2.50 rows=1 width=74)
        Index Cond: (((type)::text = 'sync'::text) AND ((sync_name)::text = (_nango_syncs.name)::text) AND (nango_config_id = 2975))
  SubPlan 1
    ->  Limit  (cost=0.86..2.88 rows=1 width=40)
          ->  Nested Loop  (cost=0.86..8570.11 rows=4245 width=40)
                ->  Index Scan using idx_jobs_syncid_createdat_where_deleted on _nango_sync_jobs  (cost=0.56..5246.20 rows=4659 width=115)
                      Index Cond: (sync_id = _nango_syncs.id)
                ->  Memoize  (cost=0.30..1.59 rows=1 width=55)
                      Cache Key: _nango_sync_jobs.sync_config_id
                      Cache Mode: logical
                      ->  Index Scan using _nango_sync_configs_pkey on _nango_sync_configs _nango_sync_configs_1  (cost=0.29..1.58 rows=1 width=55)
                            Index Cond: (id = _nango_sync_jobs.sync_config_id)
                            Filter: (NOT deleted)
```
It is slightly better even though at this point it was already quite fast

Reply:
went for your second recommendation (removing deleted = true which doesn't bring anything)

Reply:
it doesn't bring anything locally, but I'm sure in prod it's making a diff

Reply:
not a huge diff. query plan cost went down from `13.7` to `10.4`